From ff3fdc5d30ef5a253ea65c48366113627cd49dfe Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Mon, 13 Aug 2018 03:52:42 -0600
Subject: [PATCH 14/19] soong_config: Add
 TARGET_NEEDS_LEGACY_CAMERA_HAL1_DYN_NATIVE_HANDLE

* A certain subset of legacy HAL1 camera devices need to use this
  specific port mode with this specific buffer type. See
  I15f8cbb3956773e2ba97032203f5e9254505d8ac for details.

Change-Id: Ib35e14cb5cf80efc9e82a301b8e47680202f3905

# Conflicts:
#	build/soong/android/variable.go
#	build/soong/soong_config.mk
---
 build/soong/android/variable.go |  4 ++++
 build/soong/soong_config.mk     | 13 +++++++------
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index 127caf12..002527f1 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -3,6 +3,9 @@ type Product_variables struct {
 	Has_legacy_camera_hal1 struct {
 		Cflags []string
 	}
+	Needs_legacy_camera_hal1_dyn_native_handle struct {
+		Cppflags []string
+	}
 	Needs_text_relocations struct {
 		Cppflags []string
 	}
@@ -30,6 +33,7 @@ type Product_variables struct {
 type ProductVariables struct {
 	Has_legacy_camera_hal1  *bool `json:",omitempty"`
 	Java_Source_Overlays *string `json:",omitempty"`
+	Needs_legacy_camera_hal1_dyn_native_handle  *bool `json:",omitempty"`
 	Needs_text_relocations  *bool `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
 	Target_exfat_driver  *string `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 5c2a010c..4c1cbfd2 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -1,12 +1,13 @@
 _contents := $(_contents)    "Lineage":{$(newline)
 
 # See build/core/soong_config.mk for the add_json_* functions you can use here.
-$(call add_json_bool, Has_legacy_camera_hal1,                $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
-$(call add_json_str,  Java_Source_Overlays,                  $(JAVA_SOURCE_OVERLAYS))
-$(call add_json_bool, Needs_text_relocations,                $(filter true,$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS)))
-$(call add_json_str,  Specific_camera_parameter_library,     $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
-$(call add_json_str,  Target_exfat_driver,                   $(TARGET_EXFAT_DRIVER))
-$(call add_json_str,  Target_shim_libs,                      $(TARGET_LD_SHIM_LIBS))
+$(call add_json_bool, Has_legacy_camera_hal1, $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
+$(call add_json_str,  Java_Source_Overlays, $(JAVA_SOURCE_OVERLAYS))
+$(call add_json_bool, Needs_legacy_camera_hal1_dyn_native_handle, $(filter true,$(TARGET_NEEDS_LEGACY_CAMERA_HAL1_DYN_NATIVE_HANDLE)))
+$(call add_json_bool, Needs_text_relocations, $(filter true,$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS)))
+$(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
+$(call add_json_str,  Target_exfat_driver, $(TARGET_EXFAT_DRIVER))
+$(call add_json_str, Target_shim_libs, $(TARGET_LD_SHIM_LIBS))
 $(call add_json_bool, Uses_generic_camera_parameter_library, $(if $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY),,true))
 $(call add_json_bool, Uses_nvidia_enhancements,              $(filter TRUE,$(NV_ANDROID_FRAMEWORK_ENHANCEMENTS)))
 $(call add_json_bool, Uses_qcom_bsp_legacy,                  $(filter true,$(TARGET_USES_QCOM_BSP_LEGACY)))
-- 
2.17.1

